### PR TITLE
SAF Authentication Failure Auditing 

### PR DESCRIPTION
**Pull request by mmerali96**
Changes for adding z/OS SAF information to audit records.

### CODE REVIEW

File: dev/com.ibm.websphere.security/src/com/ibm/websphere/security/audit/AuditEvent.java [hunk 1]
```diff
@@ -114,6 +114,15 @@
...
     public static final String TARGET_EJB_METHOD_PARAMETERS = "target.ejb.method.parameters";
 
+    public static final String TARGET_SAF_RETURN_CODE = "target.saf.return.code";
+    public static final String TARGET_RACF_RETURN_CODE = "target.racf.return.code";
+    public static final String TARGET_RACF_REASON_CODE = "target.racf.reason.code";
+    public static final String TARGET_USER_SECURITY_NAME = "target.user.security.name";
+    public static final String TARGET_APPLID = "target.applid";
+    public static final String TARGET_SAF_PROFILE = "target.saf.profile";
+    public static final String TARGET_SAF_CLASS = "target.saf.class";
+    public static final String TARGET_AUTHORIZATION_DECISION = "target.authorization.decision";
```
Comment (kaczyns) [inline dev/com.ibm.websphere.security/src/com/ibm/websphere/security/audit/AuditEvent.java]: I guess I am questioning some of these constants... wouldn't the `target.authorization.decision` be the same as the `outcome` which is defined below?  Should we be using some standard way of indicating things like the auth decision, and the return/reason codes?  @emilytee

File: dev/com.ibm.websphere.security/src/com/ibm/websphere/security/audit/AuditEvent.java [hunk 2]
```diff
@@ -114,6 +114,15 @@
     public static final String TARGET_EJB_BEAN_NAME = "target.ejb.beanname";
     public static final String TARGET_EJB_METHOD_PARAMETERS = "target.ejb.method.parameters";
 
+    public static final String TARGET_SAF_RETURN_CODE = "target.saf.return.code";
```
Comment (kaczyns) [inline dev/com.ibm.websphere.security/src/com/ibm/websphere/security/audit/AuditEvent.java]: @emilytee Is the `target` prefix appropriate for return and reason codes that are coming back, and explain the authorization decision?

**Comment by kaczyns** [comment]
There is no personal build here, so I'm starting one now:
#build

File: dev/com.ibm.websphere.security/src/com/ibm/websphere/security/audit/AuditEvent.java (hunk 2 above)
Comment (emilytee) [reply dev/com.ibm.websphere.security/src/com/ibm/websphere/security/audit/AuditEvent.java]: Typically we put the return and reason codes in AuditEvent.REASON_CODE and AuditEvent.REASON_TYPE, and that's consistent across all of the events.  However, if there's some additional fine-grained code that is coming from SAF that is also relevant to save, you can add the target prefix to a new field,  like you did for TARGET_SAF_REASON_CODE

Comment (kaczyns) [reply dev/com.ibm.websphere.security/src/com/ibm/websphere/security/audit/AuditEvent.java]: ok thanks - I have no issues here.

**Comment by kaczyns** [comment]
well that didn't go so well... lets try again
https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_Z2_XYBtPEemsU9nTXOm12w

**Comment by mmerali96** [comment]
#build

**Comment by mmerali96** [comment]
#build

**Comment by mmerali96** [comment]
#build

**Comment by kaczyns** [comment]
I am merging this code.  The build had several failures but most of them have defects opened against them, and the ones that do not seem to be timing related (timeouts and javacores).  Given that this change is just shipping some constants to existing parts, I find it hard to believe that these changes would break the build.  I will be happy to accept responsibility if it turns out that this is the case.

**Pull request merged**